### PR TITLE
lightning: duplicate resolution support to use DELETE to cleanup clustered table

### LINKED ISSUE #44723: lightning duplication resolve support use DELETE to clean up table
## Enhancement

currently after a table is cleaned up by DELETE, lightning `duplicate-resolution = 'remove'` still see the old data and remove them

```
[2023/06/16 12:10:31.311 +08:00] [DEBUG] [duplicate.go:1076] ["[resolve-dupe] found row to resolve"] [table=`mytest`.`testtbl`] [handle=7480000000000000675F728000000000000001] [row=800002000000010201000600016161613031]
[2023/06/16 12:10:31.311 +08:00] [DEBUG] [duplicate.go:1062] ["[resolve-dupe] will delete key"] [table=`mytest`.`testtbl`] [key=7480000000000000675F728000000000000001]
[2023/06/16 12:10:31.311 +08:00] [DEBUG] [duplicate.go:1062] ["[resolve-dupe] will delete key"] [table=`mytest`.`testtbl`] [key=7480000000000000675F698000000000000001016161613031000000FC038000000000000001]
[2023/06/16 12:10:31.311 +08:00] [DEBUG] [duplicate.go:1076] ["[resolve-dupe] found row to resolve"] [table=`mytest`.`testtbl`] [handle=7480000000000000675F728000000000000001] [row=800002000000010201000600016161613031]
[2023/06/16 12:10:31.311 +08:00] [DEBUG] [duplicate.go:1062] ["[resolve-dupe] will delete key"] [table=`mytest`.`testtbl`] [key=7480000000000000675F728000000000000001]
[2023/06/16 12:10:31.311 +08:00] [DEBUG] [duplicate.go:1062] ["[resolve-dupe] will delete key"] [table=`mytest`.`testtbl`] [key=7480000000000000675F698000000000000001016161613031000000FC038000000000000001]
...
[2023/06/16 12:10:31.312 +08:00] [DEBUG] [duplicate.go:1076] ["[resolve-dupe] found row to resolve"] [table=`mytest`.`testtbl`] [handle=7480000000000000675F728000000000000001] [row=800002000000010201000600016161613031]
[2023/06/16 12:10:31.312 +08:00] [DEBUG] [duplicate.go:1062] ["[resolve-dupe] will delete key"] [table=`mytest`.`testtbl`] [key=7480000000000000675F728000000000000001]
[2023/06/16 12:10:31.312 +08:00] [DEBUG] [duplicate.go:1062] ["[resolve-dupe] will delete key"] [table=`mytest`.`testtbl`] [key=7480000000000000675F698000000000000001016161613031000000FC038000000000000001]
[2023/06/16 12:10:31.312 +08:00] [DEBUG] [duplicate.go:1076] ["[resolve-dupe] found row to resolve"] [table=`mytest`.`testtbl`] [handle=7480000000000000675F728000000000000001] [row=800002000000010201000600016161613031]
[2023/06/16 12:10:31.312 +08:00] [DEBUG] [duplicate.go:1062] ["[resolve-dupe] will delete key"] [table=`mytest`.`testtbl`] [key=7480000000000000675F728000000000000001]
[2023/06/16 12:10:31.312 +08:00] [DEBUG] [duplicate.go:1062] ["[resolve-dupe] will delete key"] [table=`mytest`.`testtbl`] [key=7480000000000000675F698000000000000001016161613031000000FC038000000000000001]
```

and the conflict row is found twice 🤔

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #44723

Problem Summary:

### What is changed and how it works?

The fix is made at TiKV side https://github.com/tikv/tikv/pull/14960

this PR fix a bug that clustered PK will generate non-clustered PK index KV and be deleted when resolve duplication

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/backend/kv/kv2sql.go [hunk 1]
```diff
@@ -88,10 +88,16 @@ func (t *TableKVDecoder) IterRawIndexKeys(h kv.Handle, rawRow []byte, fn func([]
 	}
 
 	indices := t.tbl.Indices()
+	isCommonHandle := t.tbl.Meta().IsCommonHandle
 
 	var buffer []types.Datum
 	var indexBuffer []byte
 	for _, index := range indices {
+		// skip clustered PK
+		if index.Meta().Primary && isCommonHandle {
```
Comment (D3Hunter) [inline br/pkg/lightning/backend/kv/kv2sql.go:97]: should we skip `clustered & single-int column` kind of pk?

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: br/pkg/lightning/backend/kv/kv2sql.go (hunk 1 above)
Comment (lance6716) [reply br/pkg/lightning/backend/kv/kv2sql.go:97]: added test https://github.com/pingcap/tidb/pull/44799/commits/7338cc1f191a019bbad29df2ab63d51708cf8eb5

When PKIsHandle (test file line 83), I remember the PK will not appear in `indices`, so no need to skip it.

**Review by D3Hunter — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**